qiling/extensions/r2/r2.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+#
+# Cross Platform and Multi Architecture Advanced Binary Emulation Framework
+#
+
+import functools
+import json
+import ctypes
+import libr
+from dataclasses import dataclass, fields
+from enum import Enum
+from qiling.core import Qiling
+
+@dataclass(unsafe_hash=True)
+class Function:
+    name: str
+    offset: int
+    size: int
+    signature: str
+
+    def __init__(self, **kwargs):
+        names = set([f.name for f in fields(self)])
+        for k, v in kwargs.items():
+            if k in names:
+                setattr(self, k, v)
+
+
+@dataclass(unsafe_hash=True)
+class Section:
+    name: str
+    size: int
+    vsize: int
+    paddr: int
+    vaddr: int
+    perm: str  # TODO: use int or enum
+
+    def __init__(self, **kwargs):
+        names = set([f.name for f in fields(self)])
+        for k, v in kwargs.items():
+            if k in names:
+                setattr(self, k, v)
+
+
+@dataclass(unsafe_hash=True)
+class String:
+    string: str
+    vaddr: int
+    paddr: int
+    size: int
+    length: int
+    section: str = None
+
+    def __init__(self, **kwargs):
+        names = set([f.name for f in fields(self)])
+        for k, v in kwargs.items():
+            if k in names:
+                setattr(self, k, v)
+
+
+@dataclass(unsafe_hash=True)
+class Symbol:
+    # see https://github.com/rizinorg/rizin/blob/dev/librz/include/rz_bin.h
+    class SymbolType(str, Enum):
+        NOTYPE = "NOTYPE"
+        OBJ = "OBJ"
+        FUNC = "FUNC"
+        FIELD = "FIELD"
+        IFACE = "IFACE"
+        METH = "METH"
+        STATIC = "STATIC"
+        SECT = "SECT"
+        FILE = "FILE"
+        COMMON = "COMMON"
+        TLS = "TLS"
+        NUM = "NUM"
+        LOOS = "LOOS"
+        HIOS = "HIOS"
+        LOPROC = "LOPROC"
+        HIPROC = "HIPROC"
+        SPCL = "SPCL"
+        UNK = "UNK"
+
+    class SymbolBind(str, Enum):
+        LOCAL = "LOCAL"
+        GLOBAL = "GLOBAL"
+        WEAK = "WEAK"
+        NUM = "NUM"
+        LOOS = "LOOS"
+        HIOS = "HIOS"
+        LOPROC = "LOPROC"
+        HIPROC = "HIPROC"
+        IMPORT = "IMPORT"
+        UNKNOWN = "UNKNOWN"
+
+    name: str
+    realname: str
+    bind: str
+    size: int
+    type: SymbolType
+    vaddr: int
+    paddr: int
+    is_imported: bool
+
+    def __init__(self, **kwargs):
+        names = set([f.name for f in fields(self)])
+        for k, v in kwargs.items():
+            if k in names:
+                setattr(self, k, v)
+
+
+class R2:
+    def __init__(self, ql: Qiling):
```

Comment:
Suggesting to allow the user to specify r2 evars to allow more analysis flexibility.
That could be done perhaps by accepting a dictionary of keys and values (e.g. `{'anal.hasnext' : 'true', 'anal.depth' : 5}` or any other reasonable format. Consider also allowing an "r2 init script" to let the user run a few r2 commands before starting the analysis.

Reply:
R2 has its own type of config variables, I think it's similar to qiling's profile, which is for different os though. We can extend it later.